zipkin-lens/src/actions/services-action.js

```diff
@@ -22,7 +22,9 @@ export const fetchServices = () => async (dispatch) => {
       throw Error(res.statusText);
     }
     const services = await res.json();
-    dispatch(fetchServicesSuccess(services));
+    // alphabetically sort service names since the api returns
+    // them out of order
```

Comment:
being a geek, and ignore me. Technically it isn't required to return them out of order :) it is just not required to return them in order either. iotw order is undefined.

Reply:
Involuntarily you raise a good point: this is the wrong place to fix this. I should fix it in the java API so then it's fixed for everything.

Reply:
it won't be fixed for everything because we never defined the order in the api and we know there are clones etc. iotw, it would only be "fixed" for that version of the impl. I think since the api doesn't define order, what you have is correct.

Reply:
PS I do like the idea of us sorting even if it won't obviate doing it here. When you like I think all you'd need to do is a sort here https://github.com/openzipkin/zipkin/blob/master/zipkin-server/src/main/java/zipkin2/server/internal/ZipkinQueryApiV2.java#L159 and then add a test in ITZipkinServer.  That line should cover service, span names, and tag values in one go

Reply:
I just tried that and it looks like sorting an array of strings in java is not as easy as I thought: https://circleci.com/gh/openzipkin/zipkin/3097?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
```java
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project zipkin-server: Compilation failure
[ERROR] /home/circleci/project/zipkin-server/src/main/java/zipkin2/server/internal/ZipkinQueryApiV2.java:[80,10] error: no suitable method found for sort(List<String>)
[ERROR]     method Arrays.sort(int[]) is not applicable
[ERROR]       (argument mismatch; List<String> cannot be converted to int[])
[ERROR]     method Arrays.sort(long[]) is not applicable
[ERROR]       (argument mismatch; List<String> cannot be converted to long[])
[ERROR]     method Arrays.sort(short[]) is not applicable
[ERROR]       (argument mismatch; List<String> cannot be converted to short[])
[ERROR]     method Arrays.sort(char[]) is not applicable
[ERROR]       (argument mismatch; List<String> cannot be converted to char[])
[ERROR]     method Arrays.sort(byte[]) is not applicable
[ERROR]       (argument mismatch; List<String> cannot be converted to byte[])
[ERROR]     method Arrays.sort(float[]) is not applicable
[ERROR]       (argument mismatch; List<String> cannot be converted to float[])
[ERROR]     method Arrays.sort(double[]) is not applicable
[ERROR]       (argument mismatch; List<String> cannot be converted to double[])
[ERROR]     method Arrays.sort(Object[]) is not applicable
[ERROR]       (argument mismatch; List<String> cannot be converted to Object[])
[ERROR]     method Arrays.<T#1>sort(T#1[],Comparator<? super T#1>) is not applicable
[ERROR]       (cannot infer type-variable(s) T#1
[ERROR]         (actual and formal argument lists differ in length))
[ERROR]     method Arrays.<T#2>sort(T#2[],int,int,Comparator<? super T#2>) is not applicable
[ERROR]       (cannot infer type-variable(s) T#2
[ERROR]         (actual and formal argument lists differ in length))
[ERROR]   where T#1,T#2 are type-variables:
[ERROR]     T#1 extends Object declared in method <T#1>sort(T#1[],Comparator<? super T#1>)
[ERROR]     T#2 extends Object declared in method <T#2>sort(T#2[],int,int,Comparator<? super T#2>)
```
so I dropped that.